app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -184,18 +186,35 @@ class CheckAgencyController @Inject()(
         }
       }
 
-    agentSubscriptionConnector.getRegistration(knownFacts.utr, knownFacts.postcode) flatMap {
-      case Some(Registration(Some(taxpayerName), isSubscribedToAgentServices)) if !isSubscribedToAgentServices =>
-        processCheckAgencyStatus(knownFacts.utr, taxpayerName, isSubscribedToAgentServices)
-      case Some(Registration(_, isSubscribedToAgentServices)) if isSubscribedToAgentServices =>
+    subscriptionService.getSubscriptionStatus(knownFacts.utr, knownFacts.postcode).flatMap {
+      case SubscriptionProcess(SubscriptionState.BrandNewSubscription, Some(registrationDetails)) =>
+        processCheckAgencyStatus(
+          knownFacts.utr,
+          registrationDetails.taxpayerName.get,
+          registrationDetails.isSubscribedToAgentServices)
+      case SubscriptionProcess(SubscriptionState.IsOnlySubscribedInETMP, Some(registrationDetails)) =>
+        agent match {
+          case hasNonEmptyEnrolments(_) =>
+            Future successful Redirect(routes.CheckAgencyController.showHasOtherEnrolments())
+          case _ => {
+            subscriptionService
+              .completePartialSubscription(
```

Comment:
The `completePartialSubscription` function returns a Boolean indicating whether they were partially subscribed or not.
Does this make the check above for partially subscriptions (via `getSubscriptionStatus`) now redundant, and we could just call `completePartialSubscription` without `completePartialSubscription`?

Reply:
this process wil be simplified when APB-2805 is implemented as the check will happen once, and we need to check the status to know where to redirect them.